crates/flipperzero/src/notification/mod.rs

```diff
@@ -0,0 +1,282 @@
+
+use core::ffi::c_char;
+use core::ptr;
+
+use flipperzero_sys as sys;
+use flipperzero_sys::furi::{Status, UnsafeRecord};
+
+///Default notification messages.
+pub mod messages;
+///Default notification notes.
+pub mod notes;
+///Default notification sequences.
+pub mod sequences;
+
+const RECORD_NOTIFICATION: *const c_char = sys::c_string!("notification");
+
+/// A handle to the Notification app.
+pub struct NotificationApp {
+    data: UnsafeRecord<sys::NotificationApp>,
+}
+
+impl NotificationApp {
+    /// Obtains a handle to the Notifications app.
+    pub fn open() -> Self {
+        Self {
+            data: unsafe { UnsafeRecord::open(RECORD_NOTIFICATION) },
+        }
+    }
+
+    /// Runs a notification sequence.
+    #[cfg(feature = "alloc")]
+    pub fn notify<const N: usize>(&mut self, sequence: [&'static NotificationMessage; N]) {
+        unsafe {
+            sys::notification_message(
+                self.data.as_ptr(),
+                sequence
+                    .map(|msg| match msg {
+                        NotificationMessage::End => ptr::null(),
+                        _ => Box::leak(Box::new(msg.to_sys())) as *const _,
+                    })
+                    .as_ptr() as *const _,
+            )
+        };
+    }
+}
+
+#[repr(u8)]
+#[derive(Clone, Copy)]
+pub enum Light {
+    Off = 0,
+
+    Red = 0b0001,
+    Green = 0b0010,
+    Blue = 0b0100,
+    Backlight = 0b1000,
+
+    Cyan = 0b0110,
+    Magenta = 0b0101,
+    Yellow = 0b0011,
+
+    White = 0b0111,
+}
+
+impl core::ops::BitOr for Light {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self::Output {
+        match (self, rhs) {
+            (Self::Off, l) | (l, Self::Off) => l,
+
+            (Self::Red, Self::Red) => Self::Red,
+            (Self::Green, Self::Green) => Self::Green,
+            (Self::Blue, Self::Blue) => Self::Blue,
+            (Self::Backlight, Self::Backlight) => Self::Backlight,
+            (Self::Cyan, Self::Cyan) => Self::Cyan,
+            (Self::Magenta, Self::Magenta) => Self::Magenta,
+            (Self::Yellow, Self::Yellow) => Self::Yellow,
+            (Self::White, Self::White) => Self::White,
+
+            (Self::Red, Self::Green) | (Self::Green, Self::Red) => Self::Yellow,
+            (Self::Red, Self::Blue) | (Self::Blue, Self::Red) => Self::Magenta,
+            (Self::Green, Self::Blue) | (Self::Blue, Self::Green) => Self::Cyan,
+
+            _ => Self::White, //FIXME undefined?
+        }
+    }
+}
+
+impl Light {
+    pub fn to_sys(&self) -> sys::Light {
+        match self {
+            Self::Off => 0,
+
+            Self::Red => 0b0001,
+            Self::Green => 0b0010,
+            Self::Blue => 0b0100,
+            Self::Backlight => 0b1000,
+
+            Self::Cyan => 0b0110,
+            Self::Magenta => 0b0101,
+            Self::Yellow => 0b0011,
+
+            Self::White => 0b0111,
+        }
+    }
+}
+
+/// A notification message.
+pub enum NotificationMessage {
```

Comment:
You probably want to make this a [`repr(transparent)`](https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent) newtype wrapper around `sys::NotificationMessage` and provide various `const` constructors (e.g. `NotificationMessage::delay(delay: Duration)`) to allow safely construct them.

This allows `NotificationMessage` to be directly passed to the Flipper Zero API without having to create a temporary `sys::NotificationMessage`. Otherwise a Rust enum would have been great here.